flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/UserDefinedFunctionDescriptor.java

```diff
@@ -39,12 +44,34 @@ public class UserDefinedFunctionDescriptor implements Serializable {
     private final DataType returnTypeHint;
     private final boolean isCdcPipelineUdf;
 
+    private final Map<String, String> parameters;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /** Package of {@link org.apache.flink.cdc.runtime.model.BuiltInModel}. */
+    public static final String PREFIX_CLASSPATH_MODEL = "org.apache.flink.cdc.runtime.model.";
+
+    private static final String MODEL_NAME_KEY = "name";
+
     public UserDefinedFunctionDescriptor(String name, String classpath) {
-        this.name = name;
-        this.classpath = classpath;
-        this.className = classpath.substring(classpath.lastIndexOf('.') + 1);
+        if (classpath.contains(".")) {
+            this.parameters = new HashMap<>();
+            this.name = name;
+            this.className = classpath.substring(classpath.lastIndexOf('.') + 1);
+            this.classpath = classpath;
+        } else {
+            // The UserDefinedFunction is a built-in Model.
+            try {
+                parameters = objectMapper.readValue(name, Map.class);
+            } catch (JsonProcessingException e) {
+                throw new IllegalArgumentException(e);
+            }
+            this.name = parameters.get(MODEL_NAME_KEY);
+            this.className = classpath;
+            this.classpath = PREFIX_CLASSPATH_MODEL + classpath;
+        }
```

Comment:
This is unnecessary hackaround, we can adjust `UserDefinedFunctionDescriptor` to distinguish UDFs and Models more reliably.